### ISSUE TITLE
Report empty textdomains in i18n

### ISSUE THREAD

**Issue by kkmuffme** [WordPress/WordPress-Coding-Standards#1948]
```
esc_html_e( 'foo', '' );
```

should be reported as an error, since the textdomain should not be empty (it can be, but in 99.99% of cases this is unwanted)
Same for __(), _x(),...

**Comment by jrfnl** [comment]
The `I18n` sniff will report this correctly - `Mismatched text domain. Expected 'my-plugin' but got ''.` - as soon as you either pass a text-domain via a custom ruleset or via the command-line.

This sniff is also used by WP Core and for WP Core, usage without a text-domain is valid and - as the `$text_domain` variable is optional with a default of `''`, using it like in the above code sample is too.

**Comment by kkmuffme** [comment]
Thx, had a mistake on our end not passing the textdomain correctly anymore

**Comment by jrfnl** [comment]
Looks like the default value for the `$domain` parameter is now `'default'` instead of an empty string for nearly all translation functions.

This means that passing an empty string is no longer a valid action as it would overrule the default value for the parameter and make the text string untranslatable.
